docs/rfcs/multi_locale.md

```diff
@@ -0,0 +1,159 @@
+
+Providing a good experience to allow translations of these files can be complex. In considering the UX to provide support for language specific .lg and .lu files, we should take the opportunity to consider what has become convention for how Composer bot assets are represented on the filesytem. This RFC lays out different options to write files to disk logically and proposes an update to the current convention.
+
+The distribution of .lg and .lu files in a set of Composer assets currently look like the following:
+
+```
+/ComposerDialogs
+  /common
+    common.lg
+  /Main
+    Main.dialog
+    Main.lg
+    Main.lu
+  /DialogFoo
+    DialogFoo.dialog
+    DialogFoo.lg
+    DialogFoo.lu
+  /DialogBar
+    DialogBar.dialog
+    DialogBar.lg
+    DialogBar.lu
+  /DialogBaz
+    DialogBaz.dialog
+    DialogBaz.lg
+    DialogBaz.lu
+```
+
+##### Problem
+
+We want to allow an editing experience for these files as well as allow a user to add .lg and .lu in different languages and make sensible choices on the user's behalf in how we structure the asset directory.
+
+What this would look like in today's file representation:
+
+```
+/ComposerDialogs
+  /common
+    common.en-us.lg
+    common.fr.lg
+    common.de.lg
+  /Main
+    Main.dialog
+    Main.en-us.lg
+    Main.fr.lg
+    Main.de.lg
+    Main.lu
+  /DialogFoo
+    DialogFoo.dialog
+    DialogFoo.en-us.lg
+    DialogFoo.fr.lg
+    DialogFoo.de.lg
+    DialogFoo.lu
+  /DialogBar
+    DialogBar.dialog
+    DialogBar.en-us.lg
+    DialogBar.fr.lg
+    DialogBar.de.lg
+    DialogBar.lu
+  /DialogBaz
+    DialogBaz.dialog
+    DialogBaz.en-us.lg
+    DialogBaz.fr.lg
+    DialogBaz.de.lg
+    DialogBaz.lu
+```
+
+##### Issues with current file structure
+
+Representing the .lu and .lg locally with the .dialog file is logical in that it better places the files where they are being used. This makes a Dialog directory more portable in a world where Dialogs are not only used in a single bot. This file structure is a natural place to graduate to a system where Dialogs hold their own dependencies (.lu, .lg) and can be published or shared outside of the current bot.
+
+A example downside of this approach is that this distribution of files may not be set up for domain specific work in one of the file-formats. One could prefer that all the .lg files exist in its own directly, and all the .lg files exist in its own directory, or all "en-us" files live in an "en-us" directory, and so on. Because of the anticipation of a Dialog and its associated content files (.lu, .lg) are intended to be shared via mechanisms currently planned to be built, a structure to imply a tigher binding between .dialog, .lg, .lu is currently the preferred approach.
+
+#####Note
+
+1. This proposal only applies to a filesystem-based storage plugin, and has little bearing on a database-backed store plugin implementation. **It may have merit to choose a structure that better aligns with a database-driven index approach.**
```

Comment:
It's totally OK to me that Composer only have a fs-based abstraction layer for storage, and let any other backend storage implement a few fs primitives. Similar to the idea of Unix\Linux\Plan9, everything is a file. Anyhow, i feel this is a very widely adopted approach to abstract storage and i feel no necessary to seek for a more generic storage abstraction than fs.